### PR TITLE
Bug fixes in ptg eval

### PR DESCRIPTION
**Pull request by hdefazio**
- Added option to insert interstitial, before, and finished steps into the ground truth (supports the tourniquet step-based work)
- Bug fix: there were cases in `get_time_wind_range` where the rounding of `ind1` and `ind2` caused them to be equal or `ind1 > ind2`. This meant these cases were not correctly applied to `valid` when using them to index

### CODE REVIEW

File: angel_system/data/common/discretize_data.py [hunk 1]
```diff
@@ -48,6 +50,13 @@ def get_time_wind_range(
     ind1 = max([ind1, 0])
     ind2 = min([ind2, len(time_windows)])
 
+    # assert ind2 >= ind1, "Invalid time window indexes"
+    if ind2 < ind1:
+        # Rounding issue
+        warnings.warn(
+            f"ind1: {ind1} is greater than ind2: {ind2} for start time {start} and ebd time {end}"
```
Comment (Purg) [inline angel_system/data/common/discretize_data.py:57]:
```suggestion
            f"ind1: {ind1} is greater than ind2: {ind2} for start time {start} and end time {end}"
```

File: angel_system/data/common/discretize_data.py [hunk 2]
```diff
@@ -104,19 +114,27 @@ def discretize_data_to_windows(labels, gt, detections, time_window, uncertainty_
             time_windows,
         )
 
-        valid[ind1:ind2] = True
-        correct_label = detections["class_label"][i].strip().rstrip(".")
+        correct_label = detections["class_label"][i].lower().strip().strip(".").strip()
```
Comment (Purg) [inline angel_system/data/common/discretize_data.py:117]: Why are there so many `strip` calls? Are there really class labels like `  foo  .  `? Could this not be replaced by a single `.strip(' .')`? (not sure that your class label would have newlines in there, but you could add that too if needed...)

There are more instances of this in here.

File: angel_system/data/common/load_data.py [hunk 3]
```diff
@@ -141,18 +141,21 @@ def activities_from_dive_csv(filepath: str) -> List[Activity]:
...
+
     for row in df.iterrows():
         i, s = row
         a_id = int(s[0])
         frame, time = time_from_name(s[1])
+
         if a_id not in id_to_activity:
             id_to_activity[a_id] = Activity(
-                s[9].lower().strip(),  # class label
+                s[9].lower().strip().strip(".").strip(),  # class label
```
Comment (Purg) [inline angel_system/data/common/load_data.py:158]: Same comment as above. There are more instances of this in here.

File: angel_system/data/common/load_data.py [hunk 4]
```diff
@@ -174,14 +177,76 @@ def activities_from_dive_csv(filepath: str) -> List[Activity]:
...
         f"entries: {filepath}"
     )
+
     return list(id_to_activity.values())
 
 
+def add_inter_steps(
+    gt, min_start_time, max_end_time, add_inter_steps=True, add_before_after_steps=True
+):
+    first_activity = min(gt, key=lambda a: a.start)
```
Comment (Purg) [inline angel_system/data/common/load_data.py:193]: Missing doc-string. I'm confused what this is handling.

File: angel_system/data/common/load_data.py [hunk 5]
```diff
@@ -174,14 +177,76 @@ def activities_from_dive_csv(filepath: str) -> List[Activity]:
...
         f"Some activities in source CSV do not have corresponding end time "
         f"entries: {filepath}"
     )
+
     return list(id_to_activity.values())
 
 
+def add_inter_steps(
+    gt, min_start_time, max_end_time, add_inter_steps=True, add_before_after_steps=True
+):
```
Comment (Purg) [inline angel_system/data/common/load_data.py:192]: Missing type annotations.

File: angel_system/data/common/load_data.py [hunk 6]
```diff
@@ -174,14 +177,76 @@ def activities_from_dive_csv(filepath: str) -> List[Activity]:
...
     assert np.inf not in {a.end for a in id_to_activity.values()}, (
         f"Some activities in source CSV do not have corresponding end time "
         f"entries: {filepath}"
     )
+
     return list(id_to_activity.values())
 
 
+def add_inter_steps(
+    gt, min_start_time, max_end_time, add_inter_steps=True, add_before_after_steps=True
```
Comment (Purg) [inline angel_system/data/common/load_data.py:191]: `ad_inter_steps` is both the name of the function *and* a parameter name. Consider an alternative parameter name.

File: angel_system/ptg_eval/activity_classification/evaluate_activity.py [hunk 7]
```diff
@@ -36,15 +37,39 @@ def run_eval(args):
...
         if labels is None:
             labels = l_labels
         else:
             assert labels == l_labels, (
                 f"Subsequent gt/pred pair has disjoint label sets/orders. "
                 f"{labels} != {l_labels}"
             )
+        cleaned_labels = []
+        for label in labels:
+            cleaned_labels.append(label.lower().strip().strip(".").strip())
```
Comment (Purg) [inline angel_system/ptg_eval/activity_classification/evaluate_activity.py:70]: Why is this sprinkled all over the place?

File: angel_system/data/common/discretize_data.py (hunk 2 above)
Comment (hdefazio) [reply angel_system/data/common/discretize_data.py:117]: Yeah, there were a few instances of `foo  .  `. This is mainly trying to adjust for copy & paste spacing issues that we see a lot. I've at least set it up as a function so we can adjust how the strings are cleaned in one place. Would `.strip(" . ")` work instead?

**Comment by hdefazio** [comment]
Also would like to close this MR in favor of #271 since it addresses most of these comments already

**Pull request closed without merging**